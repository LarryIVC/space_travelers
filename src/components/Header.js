import { NavLink } from 'react-router-dom';
import logo from '../assets/logo.png';

export default function Header() {
  return (
    <>
      <div className="header flex">
        <div className="brand">
          <img src={logo} alt="Logo" className="logo" width="5%" />
          <NavLink className="brand-name" to="/">space travelers&apos; hub</NavLink>
        </div>
        <div className="nav-bar flex">
          <NavLink className="nav-item" to="/">rockets</NavLink>
          <NavLink className="nav-item" to="/dragons">dragons</NavLink>
          <NavLink className="nav-item" to="/missions">missions</NavLink>
          <NavLink className="nav-item" to="/profile">my profile</NavLink>
        </div>
      </div>
    </>
  );
}
