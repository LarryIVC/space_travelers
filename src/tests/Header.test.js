import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Header component', () => {
  let header;
  beforeEach(() => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    header = screen.getByText(/Space Travelers' Hub/i);
  });

  it('Should test if Header component is inside the document', () => {
    expect(header).toBeInTheDocument();
  });

  it('Should render a snapshot of the Header component', () => {
    expect(header).toMatchSnapshot();
  });
});
