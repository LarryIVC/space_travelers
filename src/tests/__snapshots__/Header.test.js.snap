// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Header component Should render a snapshot of the Header component 1`] = `
<a
  aria-current="page"
  class="brand-name active"
  href="/"
>
  Space Travelers' Hub
</a>
`;
